2008-03-11  Jakub Jelinek  <jakub@redhat.com>

	* sysdeps/unix/sysv/linux/x86_64/lowlevellock.S
	(__lll_mutex_timedlock_wait): Use correct pointer when we don't
	call into the kernel to delay and cmpxchgl fails. Patch by
	Bryan Mason <bmason@redhat.com>.

--- libc/nptl/sysdeps/unix/sysv/linux/x86_64/lowlevellock.S.jj	2007-06-22 11:46:42.000000000 -0400
+++ libc/nptl/sysdeps/unix/sysv/linux/x86_64/lowlevellock.S	2008-03-11 06:07:39.000000000 -0400
@@ -201,7 +201,7 @@ __lll_mutex_timedlock_wait:
 
 	/* Make sure the current holder knows we are going to sleep.  */
 	movl	%edx, %eax
-	xchgl	%eax, (%rdi)
+	xchgl	%eax, (%r12)
 	testl	%eax, %eax
 	jz	6b
 	jmp	1b
